dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java

```diff
@@ -377,10 +377,12 @@ private void _saveWebAsset(Map<String, Object> contentletFormData,
 			if(UtilMethods.isSet(wfActionId)) {
 
 				WorkflowAction action = null;
-				try {
-					action = APILocator.getWorkflowAPI().findAction(wfActionId, user);
+
+				try{
+					action = APILocator.getWorkflowAPI().findActionRespectingPermissions(wfActionId, currentContentlet, user);
 				} catch(Exception e){
 
+

```

Comment:
@oswaldogallango should be remove the try?, not empty catch should exist, if the try is required and no logic in the catch should be done please add a `Log.error` but most likely the try should be remove.

Reply:
@jgambarios the try and catch can not be removed because, if the user doesn't have permission to execute that action the method is going to throw an exception but you still want to continue doing the save, but if the user have permission you just execute the fireworkflow action according to the if that is below